OpenSim/Common/Component.h

```diff
@@ -1577,10 +1595,82 @@ template <class T> friend class ComponentMeasure;
         of StateVariables by name. 
         
         NOTE: If the component name or the state variable name is ambiguous, the 
-         first instance found is returned. To disambiguate use the full name provided
+         an exception is thrown. To disambiguate use the full name provided
```

Comment:
This comment ("if ambiguous, an exception is thrown") is not entirely true: line 1618 will immediately return if `this` matches, regardless of whether ambiguity exists. Perhaps the immediate return on line 1618 is necessary for performance reasons, but it seems undesirable that the method has two potential outcomes when there's ambiguity (peaceful return if `this` happens to match, and an exception otherwise). In any event, this comment doesn't currently match the behavior of the method.